SCAutolib/models/user.py

```diff
@@ -8,52 +8,116 @@
 The classes implement add_user and delete_user methods which can be used to
 create or remove a specified user in the system or in the specified IPA server.
 """
-from pathlib import Path
+from shutil import rmtree
 
+import json
+import pwd
+import python_freeipa
+from pathlib import Path, PosixPath
+
+from SCAutolib import run, logger, LIB_DUMP_USERS, LIB_DUMP_CARDS
+from SCAutolib.exceptions import SCAutolibException
+from SCAutolib.models import card as card_model
 from SCAutolib.models.CA import IPAServerCA
-from SCAutolib import run, logger
+from SCAutolib.models.file import OpensslCnf
+
+
+class BaseUser:
+    username: str = None
+    password: str = None
+    pin: str = None
+    dump_file: Path = None
+    _cnf: OpensslCnf = None
+    _key: Path = None
+    _cert: Path = None
+    card_dir: Path = None
+    _card: card_model.Card = None
```

Comment:
Regarding naming, I don't see any reason to mention that we have some "model" there. In type `card_model` is used as alias to avoid collision with class property
```python3
from SCAutolib.models import card as card_model
```
VS
```python3
@property
def card(self):
    ...
```
Does it make sense for you to leave just `_card` name?